openadapt/config.py

```diff
@@ -174,18 +178,54 @@ def obfuscate(val: str, pct_reveal: float = 0.1, char: str = "*") -> str:
             logger.info(f"{key}={val}")
 
 
-def filter_log_messages(data: dict) -> bool:
-    """Filter log messages by ignoring specific strings.
-
-    Args:
-        data (dict): Data from a loguru logger.
-
-    Returns:
-        bool: True if the message should not be ignored, False if it should be ignored.
-    """
-    # TODO: ultimately, we want to fix the underlying issues, but for now,
-    # we can ignore these messages
-    messages_to_ignore = [
-        "Cannot pickle Objective-C objects",
-    ]
-    return not any(msg in data["message"] for msg in messages_to_ignore)
+class LogMessageFilter:
+    """Filter log messages, excluding specific strings and limiting warnings."""
+
+    def __init__(
+        self,
+        max_num_warnings_per_second: int = MAX_NUM_WARNINGS_PER_SECOND,  # noqa: F821
+    ) -> None:
+        """
+        Initialize a new instance of LogMessageFilter.
+
+        Args:
+            max_num_warnings_per_second: The maximum number of warnings allowed
+                per second. Defaults to the value of MAX_NUM_WARNINGS_PER_SECOND.
+        """
+        self.max_num_warnings_per_second = max_num_warnings_per_second
+        # TODO: ultimately, we want to fix the underlying issues, but for now,
+        # we can ignore these messages
+        self.messages_to_ignore = ["Cannot pickle Objective-C objects"]
```

Comment:
What do you think about moving this to `config`?

Also, what do you think about moving this to `openadapt/logging.py`? Seems like that's where it belongs rather than in `config.py`, and then we can use the module as a namespace instead of defining a class.

Reply:
resolved in https://github.com/OpenAdaptAI/OpenAdapt/pull/389/commits/344e8881df03bf7c22b2c103a111a09f91a8eebd